### PR TITLE
Improved docs resample

### PR DESCRIPTION
**Pull request by jpaasen**
This fixes https://github.com/librosa/librosa/issues/1613

### CODE REVIEW

**Review by bmcfee — APPROVED** [review]
LGTM, thanks!

File: librosa/core/audio.py [hunk 1]
```diff
@@ -527,22 +527,27 @@ def resample(
...
+            `soxr` Quick cubic interpolation (very fast, but not bandlimited)
         'kaiser_best'
             `resampy` high-quality mode
         'kaiser_fast'
             `resampy` faster method
         'fft' or 'scipy'
             `scipy.signal.resample` Fourier method.
         'polyphase'
-            `scipy.signal.resample_poly` polyphase filtering. (fast)
+            `scipy.signal.resample_poly` polyphase filtering. (fast, but not bandlimited)
```
Comment (bmcfee) [inline librosa/core/audio.py:538]: actually, not sure this is correct?  The scipy documentation explicitly states that the signal is low-pass filtered: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.resample_poly.html

Comment (jpaasen) [reply librosa/core/audio.py:538]: Might be, but it doesn't look like a good one.

image

Comment (jpaasen) [reply librosa/core/audio.py:538]: Ref my colab: https://colab.research.google.com/drive/1VNS8O494CYocHrilUc7lZ_hKltNqb_PG?usp=sharing#scrollTo=4D3_oaecydbH

Comment (bmcfee) [reply librosa/core/audio.py:538]: Oh indeed, it's not a *good* filter, but it's qualitatively different from something like zero-order hold.

Comment (jpaasen) [reply librosa/core/audio.py:538]: Yes. With linear you get a clean aliased pyramid :-)

image

Comment (jpaasen) [reply librosa/core/audio.py:538]: Should I remove the `but not bandlimited` completly for `polyphase`, or change it to something like `slightly bandlimited`?

Comment (jpaasen) [reply librosa/core/audio.py:538]: It will still be covered by the `note` section.

File: librosa/core/audio.py [hunk 2]
```diff
@@ -537,11 +537,15 @@ def resample(
...
             `scipy.signal.resample_poly` polyphase filtering. (fast)
         'linear'
-            `samplerate` linear interpolation. (very fast)
+            `samplerate` linear interpolation. (very fast, but not bandlimited)
         'zero_order_hold'
-            `samplerate` repeat the last value between samples. (very fast)
+            `samplerate` repeat the last value between samples. (very fast, but not bandlimited)
         'sinc_best', 'sinc_medium' or 'sinc_fastest'
-            `samplerate` high-, medium-, and low-quality sinc interpolation.
+            `samplerate` high-, medium-, and low-quality bandlimited sinc interpolation.
```
Comment (jpaasen) [inline librosa/core/audio.py:544]: I also added `bandlimited` here for consistency. It is also stated explicitly in the `libsamplerate` doc: http://www.mega-nerd.com/SRC/api_misc.html

**Review by bmcfee — APPROVED** [review]

**Comment by bmcfee** [comment]
It looks like the docstring linter (velin) is complaining about trailing spaces.  This isn't easy to see from the github actions render of the log, but the raw log dump shows it:

```
-            Not all options yield a bandlimited interpolator. If you use `soxr_qq`, `polyphase`, [39;49;00m[37m[39;49;00m
+            Not all options yield a bandlimited interpolator. If you use `soxr_qq`, `polyphase`,[39;49;00m[37m[39;49;00m
```

**Comment by jpaasen** [comment]
The lint failure looks related to my changes, but the output is not really descriptive.

**Comment by bmcfee** [comment]
Right - your changes have spaces at the end of the line that should be removed.  The way github actions renders log output makes this impossible to see, but it's visible in the quote I pasted above.

**Comment by jpaasen** [comment]
The failed 3.10 builds look unrelated to these changes.

**Review by bmcfee — APPROVED** [review]

**Comment by bmcfee** [comment]
> The failed 3.10 builds look unrelated to these changes.

Agreed - codecov has been failing randomly for a bit and I haven't had time to look into it.  This otherwise LGTM, thanks again!

**Comment by jpaasen** [comment]
You're welcome. Happy to contribute.

**Pull request merged**